server/src/transport/ws.rs

```diff
@@ -426,16 +444,26 @@ where
 }
 
 /// Attempts to read data from WebSocket fails if the server was stopped.
-async fn try_recv<S>(receiver: &mut Receiver, data: &mut Vec<u8>, stopped: S) -> Receive<S>
+async fn try_recv<S>(
+	receiver: &mut Receiver,
+	data: &mut Vec<u8>,
+	mut stopped: S,
+	max_inactive_interval: Duration,
+	disconnect_notify: Arc<Notify>,
+) -> Receive<S>
 where
 	S: Future<Output = ()> + Unpin,
 {
+	let last_active = Mutex::new(Instant::now());
 	let receive = async {
 		// Identical loop to `soketto::receive_data` with debug logs for `Pong` frames.
 		loop {
 			match receiver.receive(data).await? {
 				soketto::Incoming::Data(d) => break Ok(d),
```

Comment:
I think it makes sense to take `data messages` into account as well because it possible if some client attempts to send lots of messages the pings may be delayed....